runtime/runtime/src/lib.rs

```diff
@@ -1568,6 +1568,84 @@ impl Runtime {
         Ok(local_receipts)
     }
 
+    /// This function wraps [Runtime::process_receipt]. It adds a tracing span around the latter
+    /// and populates various metrics.
+    fn process_receipt_with_metrics(
+        &self,
+        receipt: &Receipt,
+        state_update: &mut TrieUpdate,
+        total: &mut TotalResourceGuard,
+        protocol_version: ProtocolVersion,
+        apply_state: &ApplyState,
+        epoch_info_provider: &dyn EpochInfoProvider,
+        outcomes: &mut Vec<ExecutionOutcomeWithId>,
+        mut stats: &mut ApplyStats,
+        mut receipt_sink: &mut ReceiptSink,
+        mut validator_proposals: &mut Vec<ValidatorStake>,
```

Comment:
ditto on order of arguments - another plus is that it would be somewhat stable in different methods -again totally optional